### DISCUSSION TITLE
io.popen call to curl pollutes the buffer when it fails

### DISCUSSION THREAD

**Discussion by sidequestboy** [oflisback/obsidian-bridge.nvim#3] [Ideas]
Hi thanks for the plugin! I'm trying this out on macos, and encountered 

<img width="842" alt="Screenshot 2023-09-20 at 2 08 27 PM" src="https://github.com/oflisback/obsidian-sync.nvim/assets/1538657/ea709e71-c960-4ff6-8952-7f41873027f0">

when switching buffers from my obsidian vault. The problem was related to spaces in my filenames, which is what #1 is about.

This issue is about not handling the subprocess stdio streams.

Using lua's native `io.popen` for the `curl` call sends stderr to the cursor position I believe, as described in [this open neovim issue](https://github.com/neovim/neovim/issues/21376)

Fixing the issue might be a simple as appending `>/dev/null 2>&1` to the `popen` call instead of `>/dev/null`, but perhaps it would be good to look at [plenary.nvim's Job](https://github.com/nvim-lua/plenary.nvim#plenaryjob) for async subprocess handling (seems simple), or `uv.spawn()` (more complex).

I actually did give this a shot today using both of those methods though, and neither actually worked properly - I'm not sure what I was missing, but I could share that code in another PR if you're interested.

**Comment by oflisback** [comment]
Thank you for the thorough investigation! With uriencoding being added with https://github.com/oflisback/obsidian-sync.nvim/commit/6368d1b154a49d9e17bc71cfbc8bf9c2a6b78457 (or similar) I think it'd be interesting to see another scenario that results in buffer pollution to understand if it's worth fixing. Perhaps another alternative would be to try out  plenary's curl wrapper https://github.com/nvim-lua/plenary.nvim/blob/master/lua/plenary/curl.lua.

**Comment by sidequestboy** [comment]
oh that might be a good solution, also I noticed that wrapper includes a urlencode function as a bonus! Also, as far as neovim dependencies go, plenary is fairly ubiquitous already.

**Comment by sidequestboy** [comment]
epwalsh/obsidian.nvim already depends on plenary in particular

**Comment by oflisback** [comment]
Yes I really like the idea and plenary feels like such an ubiquitous dependency. The current approach to curl doesn't allow extraction of HTTP response code for instance, which would be very handy. With the new `scroll_sync` option it would for example be nice to look for HTTP 404 and tell the user that it's trying to use a feature that the installed version of Local REST API obsidian plugin doesn't support.
